src/combinatorics/simple_graph/basic.lean

```diff
@@ -332,4 +332,41 @@ finset.max' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty
 
 end finite
 
+section complement
+
+/-!
+## Complement of a simple graph
+
+This section contains definitions and lemmas concerning the complement of a simple graph.
+-/
+
+/--
+We define `complement G` to be the `simple_graph V` such that no two adjacent vertices in `G`
+are adjacent in the complement, and every nonadjacent pair of vertices is adjacent
+(still ensuring that vertices are not adjacent to themselves.)
+-/
+def complement (G : simple_graph V) : simple_graph V :=
+{ adj := λ v w, v ≠ w ∧ ¬G.adj v w,
+  sym := λ v w ⟨hne, _⟩, ⟨hne.symm, by rwa edge_symm⟩,
+  loopless := λ v ⟨hne, _⟩, false.elim (hne rfl) }
+
+/--
+The complement of the complement of `G` is equal to `G`.
+-/
+lemma complement_involutive (G : simple_graph V) : complement (complement G) = G :=
```

Comment:
I'm not exactly sure the naming convention (@eric-wieser seems to pop in with comments about involution lemmas, so I'll ping him), but maybe this is `compl_invol` and then you can have
```lean
lemma compl_involutive : function.involutive (@compl V) := sorry
```
as well

Reply:
I think the name of this lemma is correct as is (eg list.reverse_reverse), but adding the `function.involutive` version as well is a nice idea